### PR TITLE
Tumbleweed wallpaper refresh

### PR DESCRIPTION
**Pull request by lkocman**
Unified 4096x4096 TW wallpaper for Tumbleweed
 
* Graphics from @hellcp. Many thanks for all the work!
* Resolves TW part of gh#openSUSE/branding#135
* Covers both dark and light variant
* Use single 4096x4096 file as mentioned in gh#openSUSE/branding#161
* Use the dark 800x600 also for the back.png
* Use default.png instead of default-light.png to maximize compatability.

screenshot

### CODE REVIEW

**Comment by lkocman** [comment]
Here's the transition from light to dark variant and show of matching overlap as advised by @hellcp 
[wallpaper-transition.webm](https://github.com/user-attachments/assets/f726b83f-b8fc-44af-bce4-44bc1b396d7a)

**Comment by lkocman** [comment]
I'm thinking if we really have to generate all individual resolutions or just the largest one and scale down (gnome has background scale properties etc). Also based on checking https://help.gnome.org/admin/system-admin-guide/stable/backgrounds-extra.html.en it seems that our xml si bit dated (backgrounds vs walllpaper).

Do we want automatic transition or rather keep uri/dark-uri  options (without time-based change).

If we'd go for time-based change then we can't have the distribution one as default, as that would kill openQA. There I'd advise project. But I think sticking with light/dark works.

**Comment by hellcp** [comment]
Why would we do time based change? Let's change them based on light or dark theme. openQA to my knowledge doesn't test dark theme, so it wouldn't make any difference to it, since it would only ever see the light theme wallpaper. Also I get the feeling that project wallpaper would fit better in the wallpapers repository since it's presumably also going to be shared between Leap and Tumbleweed

**Comment by hellcp** [comment]
Screenshot from 2024-09-24 02-29-59
I haven't investigated it deeply, but why do they look this different?

**Comment by lkocman** [comment]
> Why would we do time based change? Let's change them based on light or dark theme. openQA to my knowledge doesn't test dark theme, so it wouldn't make any difference to it, since it would only ever see the light theme wallpaper. Also I get the feeling that project wallpaper would fit better in the wallpapers repository since it's presumably also going to be shared between Leap and Tumbleweed

Fully agreed about dark/light theme @hellcp.

The only problem with wallpapers is that it will very likely be just suggested. So we don't dramatically increase space on livecds etc. And then the project one would not make it to many systems.

I'm fine with the digital-art/geo wallpaper + photo be in a suggested walllpaper but I really wanted to hhave these two (three if we count dark/light separately) on each system.

**Comment by lkocman** [comment]
> Screenshot from 2024-09-24 02-29-59 I haven't investigated it deeply, but why do they look this different?

Good catch @hellcp

**Comment by lkocman** [comment]
> > Screenshot from 2024-09-24 02-29-59 I haven't investigated it deeply, but why do they look this different?
> 
> Good catch @hellcp

Must be a github glitch where they display rendered png vs svg. I think we should investigate why the png is so much lighter.

here's png export 16:9 vs 16:10

image

Here are svgs in inkscape

image

**Comment by lkocman** [comment]
Let me make a test build to avoid any issues in makefile etc ...

**Comment by lkocman** [comment]
Make passed, ready to get merged!

```📦[lkocman@osc branding]$ make
echo "Make sure to have rsvg-view and GraphicsMagick installed"
Make sure to have rsvg-view and GraphicsMagick installed
mkdir -p ~/.fonts openSUSE/gfxboot/data-boot/ openSUSE/gfxboot/data-install
cp gfxboot/SourceSansPro-Light.ttf ~/.fonts
for name in back welcome on off glow; do \
	rsvg-convert raw-theme-drop/${name}.svg -w 800 -a -o tmp-gfxboot.d.png; \
	gm convert -quality 100 -interlace None -colorspace YCbCr -geometry 800x600 -sampling-factor 2x2 tmp-gfxboot.d.png openSUSE/gfxboot/data-install/${name}.jpg; \
	rm tmp-gfxboot.d.png; \
done
rsvg-convert gfxboot/text.svg -w 114 -a -o tmp-gfxboot.d.png
gm convert -quality 100 -interlace None -colorspace YCbCr -sampling-factor 2x2 tmp-gfxboot.d.png openSUSE/gfxboot/data-install/text.jpg
rm tmp-gfxboot.d.png
rsvg-convert raw-theme-drop/back.svg -w 800 -a -o tmp-gfxboot.d.png
gm convert -quality 100 -interlace None -colorspace YCbCr -geometry 800x600 -sampling-factor 2x2 tmp-gfxboot.d.png openSUSE/gfxboot/data-boot/back.jpg
rm tmp-gfxboot.d.png
rm ~/.fonts/SourceSansPro-Light.ttf
mkdir -p openSUSE/gnome
sed "s:@VERSION@:tumbleweed:g;s:@GNOME_STATIC_DYNAMIC@:static:g" gnome/wallpaper-branding-openSUSE.xml.in > openSUSE/gnome/wallpaper-branding-openSUSE.xml
cp gnome/openSUSE-default-static.xml openSUSE/gnome/openSUSE-default-static.xml
mkdir -p openSUSE/grub2
cp -a boot/grub2/theme openSUSE/grub2/
rm -rf openSUSE/icewm
mkdir -p openSUSE/icewm
cp -av icewm openSUSE/
'icewm/theme' -> 'openSUSE/icewm/theme'
'icewm/themes' -> 'openSUSE/icewm/themes'
'icewm/themes/yast-installation' -> 'openSUSE/icewm/themes/yast-installation'
'icewm/themes/yast-installation/LICENSE' -> 'openSUSE/icewm/themes/yast-installation/LICENSE'
'icewm/themes/yast-installation/closeA.xpm' -> 'openSUSE/icewm/themes/yast-installation/closeA.xpm'
'icewm/themes/yast-installation/closeO.xpm' -> 'openSUSE/icewm/themes/yast-installation/closeO.xpm'
'icewm/themes/yast-installation/default.theme' -> 'openSUSE/icewm/themes/yast-installation/default.theme'
'icewm/themes/yast-installation/frameAB.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameAB.xpm'
'icewm/themes/yast-installation/frameABL.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameABL.xpm'
'icewm/themes/yast-installation/frameABR.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameABR.xpm'
'icewm/themes/yast-installation/frameAL.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameAL.xpm'
'icewm/themes/yast-installation/frameAR.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameAR.xpm'
'icewm/themes/yast-installation/frameAT.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameAT.xpm'
'icewm/themes/yast-installation/frameATL.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameATL.xpm'
'icewm/themes/yast-installation/frameATR.xpm' -> 'openSUSE/icewm/themes/yast-installation/frameATR.xpm'
'icewm/themes/yast-installation/icons' -> 'openSUSE/icewm/themes/yast-installation/icons'
'icewm/themes/yast-installation/icons/app_11x11.xpm' -> 'openSUSE/icewm/themes/yast-installation/icons/app_11x11.xpm'
'icewm/themes/yast-installation/icons/folder_11x11.xpm' -> 'openSUSE/icewm/themes/yast-installation/icons/folder_11x11.xpm'
'icewm/themes/yast-installation/icons/xterm.xpm' -> 'openSUSE/icewm/themes/yast-installation/icons/xterm.xpm'
'icewm/themes/yast-installation/maximizeA.xpm' -> 'openSUSE/icewm/themes/yast-installation/maximizeA.xpm'
'icewm/themes/yast-installation/maximizeO.xpm' -> 'openSUSE/icewm/themes/yast-installation/maximizeO.xpm'
'icewm/themes/yast-installation/menuButtonA.xpm' -> 'openSUSE/icewm/themes/yast-installation/menuButtonA.xpm'
'icewm/themes/yast-installation/menubg.xpm' -> 'openSUSE/icewm/themes/yast-installation/menubg.xpm'
'icewm/themes/yast-installation/menusel.xpm' -> 'openSUSE/icewm/themes/yast-installation/menusel.xpm'
'icewm/themes/yast-installation/minimizeA.xpm' -> 'openSUSE/icewm/themes/yast-installation/minimizeA.xpm'
'icewm/themes/yast-installation/minimizeO.xpm' -> 'openSUSE/icewm/themes/yast-installation/minimizeO.xpm'
'icewm/themes/yast-installation/taskbar' -> 'openSUSE/icewm/themes/yast-installation/taskbar'
'icewm/themes/yast-installation/taskbar/collapse.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/collapse.xpm'
'icewm/themes/yast-installation/taskbar/desktop.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/desktop.xpm'
'icewm/themes/yast-installation/taskbar/expand.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/expand.xpm'
'icewm/themes/yast-installation/taskbar/linux_backup.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/linux_backup.xpm'
'icewm/themes/yast-installation/taskbar/start.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/start.xpm'
'icewm/themes/yast-installation/taskbar/taskbarbg.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/taskbarbg.xpm'
'icewm/themes/yast-installation/taskbar/taskbuttonactive.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/taskbuttonactive.xpm'
'icewm/themes/yast-installation/taskbar/windows.xpm' -> 'openSUSE/icewm/themes/yast-installation/taskbar/windows.xpm'
'icewm/themes/yast-installation/titleAB.xpm' -> 'openSUSE/icewm/themes/yast-installation/titleAB.xpm'
mkdir -p openSUSE/libreoffice/program
cp -r libreoffice/flat_logo.svg libreoffice/sofficerc libreoffice/shell openSUSE/libreoffice/program/
rsvg-convert libreoffice/intro.svg -o openSUSE/libreoffice/program/intro.png
mkdir -p openSUSE/wallpapers openSUSE/wallpapers/openSUSEdefault/contents/images
rsvg-convert raw-theme-drop/default-dark.svg -o openSUSE/wallpapers/openSUSEdefault/contents/images/default-dark.png
rsvg-convert raw-theme-drop/default.svg -o openSUSE/wallpapers/openSUSEdefault/contents/images/default.png
optipng -o5 openSUSE/wallpapers/openSUSEdefault/contents/images/default-dark.png
** Processing: openSUSE/wallpapers/openSUSEdefault/contents/images/default-dark.png
4096x4096 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 471502 bytes
Input file size = 472261 bytes

Trying:
  zc = 9  zm = 9  zs = 0  f = 2		IDAT size = 453518
  zc = 9  zm = 8  zs = 0  f = 2		IDAT size = 453260
  zc = 9  zm = 9  zs = 1  f = 2		IDAT size = 450340
  zc = 9  zm = 8  zs = 1  f = 2		IDAT size = 450067
  zc = 9  zm = 9  zs = 3  f = 5		IDAT size = 431112
                               
Selecting parameters:
  zc = 9  zm = 9  zs = 3  f = 5		IDAT size = 431112

Output IDAT size = 431112 bytes (40390 bytes decrease)
Output file size = 431187 bytes (41074 bytes = 8.70% decrease)

optipng -o5 openSUSE/wallpapers/openSUSEdefault/contents/images/default.png
** Processing: openSUSE/wallpapers/openSUSEdefault/contents/images/default.png
4096x4096 pixels, 3x8 bits/pixel, RGB
Input IDAT size = 210183 bytes
Input file size = 210558 bytes

Trying:
  zc = 9  zm = 9  zs = 0  f = 2		IDAT size = 207655
  zc = 9  zm = 8  zs = 0  f = 2		IDAT size = 206931
  zc = 9  zm = 9  zs = 1  f = 2		IDAT size = 203789
  zc = 9  zm = 8  zs = 1  f = 2		IDAT size = 203105
  zc = 9  zm = 9  zs = 3  f = 5		IDAT size = 188331
  zc = 9  zm = 8  zs = 3  f = 5		IDAT size = 187939

Selecting parameters:
  zc = 9  zm = 8  zs = 3  f = 5		IDAT size = 187939

Output IDAT size = 187939 bytes (22244 bytes decrease)
Output file size = 188014 bytes (22544 bytes = 10.71% decrease)

cp -p kde-workspace/metadata.json openSUSE/wallpapers/openSUSEdefault/metadata.json
cp -p raw-theme-drop/screenshot.png openSUSE/wallpapers/openSUSEdefault/screenshot.png
cp -p kde-workspace/metadata.json openSUSE/wallpapers/openSUSEdefault/metadata.json
mkdir -p openSUSE/yast_wizard
cp -a yast/* openSUSE/yast_wizard
mkdir -p openSUSE/plymouth
cp plymouth/config/plymouthd.defaults openSUSE/plymouth
📦[lkocman@osc branding]$ 
```

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,7 @@ THEME=openSUSE
 all: info openSUSE.d
 
 info:
-	echo "Make sure to have rsvg-view and GraphicsMagick installed"
+	echo "Make sure to have rsvg-view, GraphicsMagick and optipng installed"
```
Comment (lkocman) [inline Makefile:8]: Sync with Leap 16.0, we do use optipng

File: Makefile [hunk 2]
```diff
@@ -67,25 +67,26 @@ libreoffice.d:
 	mkdir -p openSUSE/libreoffice/program
 	cp -r libreoffice/flat_logo.svg libreoffice/sofficerc libreoffice/shell openSUSE/libreoffice/program/
 	rsvg-convert libreoffice/intro.svg -o openSUSE/libreoffice/program/intro.png
+	optipng -o7 openSUSE/libreoffice/program/intro.png
```
Comment (lkocman) [inline Makefile:70]: Sync with Leap 16.0 we are using optipng there too

File: gnome/openSUSE-default-static.xml [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
+<?xml version="1.0"?>
+<!DOCTYPE wallpapers SYSTEM "gnome-wp-list.dtd">
+<wallpapers>
+  <wallpaper deleted="false">
+    <name>Default Background</name>
+    <filename>/usr/share/wallpapers/openSUSEdefault/contents/images/default.png</filename>
+    <filename-dark>/usr/share/wallpapers/openSUSEdefault/contents/images/default-dark.png</filename-dark>
+    <options>zoom</options>
+    <shade_type>solid</shade_type>
+    <pcolor>#63bbb2ff</pcolor>
```
Comment (lkocman) [inline gnome/openSUSE-default-static.xml:10]: Used Tumbleweed light bg color as primary, dark bg as secondary

**Comment by lkocman** [comment]
Shared with Marketing channel, no feedback in PR, seems like people are okay with it. Merging!

**Pull request merged**